plugins/googleanalytics/src/org/pentaho/di/trans/steps/googleanalytics/GoogleAnalyticsApiFacade.java

```diff
@@ -0,0 +1,104 @@
+ *
+ * Copyright 2002 - 2015 Pentaho Corporation (Pentaho). All rights reserved.
+ *
+ * NOTICE: All information including source code contained herein is, and
+ * remains the sole property of Pentaho and its licensors. The intellectual
+ * and technical concepts contained herein are proprietary and confidential
+ * to, and are trade secrets of Pentaho and may be covered by U.S. and foreign
+ * patents, or patents in process, and are protected by trade secret and
+ * copyright laws. The receipt or possession of this source code and/or related
+ * information does not convey or imply any rights to reproduce, disclose or
+ * distribute its contents, or to manufacture, use, or sell anything that it
+ * may describe, in whole or in part. Any reproduction, modification, distribution,
+ * or public display of this information without the express written authorization
+ * from Pentaho is strictly prohibited and in violation of applicable laws and
+ * international treaties. Access to the source code contained herein is strictly
+ * prohibited to anyone except those individuals and entities who have executed
+ * confidentiality and non-disclosure agreements or other agreements with Pentaho,
+ * explicitly covering such access.
+ */
+
+/* Copyright (c) 2008 Google Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package org.pentaho.di.trans.steps.googleanalytics;
+
+
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.services.analytics.Analytics;
+import com.google.api.services.analytics.AnalyticsScopes;
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import org.pentaho.di.core.util.Assert;
+
+import java.io.File;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+
+
+public class GoogleAnalyticsApiFacade {
+
+  private Analytics analytics;
+
+  public static GoogleAnalyticsApiFacade createFor( String application, GaInputStepMeta meta )
+    throws IOException, GeneralSecurityException {
+
+    return new GoogleAnalyticsApiFacade(
+      GoogleNetHttpTransport.newTrustedTransport(),
+      JacksonFactory.getDefaultInstance(),
+      application,
+      meta.getOAuthServiceAccount(),
+      new File( meta.getOAuthKeyFile() ) );
```

Comment:
Neet environmentSubstitute here (or somewhere close by) ... as the transformation moves from place to place, the key file will be in different places for sure.